src/lib/providers/ModalProvider/ModalTypes.ts

```diff
@@ -0,0 +1,34 @@
+//- Containers & Components Imports
+import { MintNewNFTProps } from 'containers/flows/MintNewNFT/MintNewNFT';
+
+export enum ModalType {
+	NULL_MODAL_TYPE,
+	MINT_NEW_NFT,
+	CONNECT_TO_WALLET,
+}
```

Comment:
I am not sure if we need `NULL_MODAL_TYPE `. And I think it will be better to add value of enum type.

Also, I was going to create `modal` redux state and `hooks`. But you created provider already. Anyway, I have to switch this `context provider` to redux later.

I have question, do you know how many `Modal`s do we have in the application? I think it is good to know the amount of modals and includes all in this PR.